### PR TITLE
Alerting: Add new button for exporting new alert rule in HCL format

### LINKED ISSUE #96320: Grafana Alerting: Cannot add alert rule to a provisioned alert group
### What happened?

A customer has a provisioned alert group that they are trying to add a new alert rule to using TF. They are getting an error that they cannot add the rule as it's a provisioned or "locked down rule". We should ensure users can use TF to add rules to existing provisioned alert groups. 

### What did you expect to happen?

As TF is a provisioning / as code method, the user should be able to add a new alert rule into a provisioned alert group via terraform

### Did this work before?

unsure if this ever worked.

### How do we reproduce it?

1. make an alert group with an alert through the prov API
2. Try and add a new rule to that group via TF
3.

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: 11.4 - Cloud
OS:
Browser:

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR adds a new button in the alert list view to export a new alert rule (not saved yet). 
Currently, this new export will be limited to `HCL` format since handling the new `UIDs` required for other provisioning methods is still under discussion. For Terraform, UIDs are not necessary for provisioning.

**Why do we need this feature?**

Some users need to export a rule that does not exist yet

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/96320

**Special notes for your reviewer:**

https://github.com/user-attachments/assets/330d619a-6c40-4ef1-b54e-601a34e58458

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/RuleList.test.tsx [hunk 1]
```diff
@@ -120,7 +120,7 @@ const ui = {
   rulesFilterInput: byTestId('search-query-input'),
   moreErrorsButton: byRole('button', { name: /more errors/ }),
   editCloudGroupIcon: byTestId('edit-group'),
-  newRuleButton: byText(/new alert rule/i),
+  newRuleButton: byRole('link', { name: 'New alert rule' }),
```
Comment (konrad147) [inline public/app/features/alerting/unified/RuleList.test.tsx:123]: ❤️

File: public/app/features/alerting/unified/components/export/ExportNewGrafanaRule.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  children: React.ReactNode;
+}
+
+function ExportNewGrafanaRuleWrapper({ children }: ExportNewGrafanaRuleWrapperProps) {
+  return (
+    <AlertingPageWrapper
+      navId="alert-list"
+      pageNav={{
+        text: 'Export new Grafana rule',
+        subTitle: 'Export a new rule definition in the format of your choice. Any changes you make will not be saved.',
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/export/ExportNewGrafanaRule.tsx:24]: Should we adjust the message and be explicit that only Terraform is supported?
Like `Export a new rule definition in Terraform(HCL) format. Any changes....`?

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/ModifyExportRuleForm.tsx [hunk 3]
```diff
@@ -129,12 +146,12 @@ interface GrafanaRuleDesignExportPreviewProps {
...
   exportValues: RuleFormValues;
-  uid: string;
+  uid?: string;
 }
 export const getPayloadToExport = (
-  uid: string,
   formValues: RuleFormValues,
-  existingGroup: RulerRuleGroupDTO<RulerRuleDTO> | null | undefined
+  existingGroup: RulerRuleGroupDTO<RulerRuleDTO> | null | undefined,
+  uid?: string
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/ModifyExportRuleForm.tsx:154]:
```suggestion
  ruleUid?: string
```

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/ModifyExportRuleForm.tsx [hunk 4]
```diff
@@ -167,10 +184,10 @@ export const getPayloadToExport = (
   }
 };
 
-const useGetPayloadToExport = (values: RuleFormValues, uid: string) => {
+const useGetPayloadToExport = (values: RuleFormValues, uid?: string) => {
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/ModifyExportRuleForm.tsx:187]:
```suggestion
const useGetPayloadToExport = (values: RuleFormValues, ruleUid?: string) => {
```

File: public/app/features/alerting/unified/rule-list/RuleList.v1.tsx [hunk 5]
```diff
@@ -169,3 +180,21 @@ export function CreateAlertButton() {
...
+  const url = createRelativeUrl(`/alerting/export-new-rule`, {
+    returnTo,
+  });
+  return (
+    <LinkButton
+      href={url}
+      icon="download-alt"
+      variant="secondary"
+      tooltip="Export new grafana rule"
+      onClick={() => logInfo(LogMessages.grafanaRecording)}
```
Comment (konrad147) [inline public/app/features/alerting/unified/rule-list/RuleList.v1.tsx:195]: why we log `grafanaRecording`?

**Review by konrad147 — APPROVED** [review]
Code LGTM! a few non-blocking nits

File: public/app/features/alerting/unified/rule-list/RuleList.v1.tsx (hunk 5 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/rule-list/RuleList.v1.tsx:195]: ops, wrong one

**Pull request merged**